tests/integration/tier0/custom-kernel/test_cutsom_kernel.py

```diff
@@ -0,0 +1,82 @@
+import os
+import platform
+
+import pytest
+
+
+DISTRO_KERNEL_MAPPING = {
+    "centos-7": {
+        "yum_install_cmd": "yum install https://yum.oracle.com/repo/OracleLinux/OL7/latest/x86_64/getPackage/kernel-3.10.0-1160.76.1.0.1.el7.x86_64.rpm -y",
+        "grub_cmd": "grub2-set-default 'CentOS Linux (3.10.0-1160.76.1.0.1.el7.x86_64) 7 (Core)'",
+    },
+    "centos-8.4": {
+        "yum_install_cmd": "yum install https://yum.oracle.com/repo/OracleLinux/OL8/4/baseos/base/x86_64/getPackage/kernel-core-4.18.0-305.el8.x86_64.rpm -y",
+        "grub_cmd": "grub2-set-default 'Oracle Linux Server (4.18.0-305.el8.x86_64) 8.4'",
+    },
+    "centos-8.5": {
+        "yum_install_cmd": "yum install https://yum.oracle.com/repo/OracleLinux/OL8/5/baseos/base/x86_64/getPackage/kernel-core-4.18.0-348.el8.x86_64.rpm -y",
+        "grub_cmd": "grub2-set-default 'Oracle Linux Server (4.18.0-348.el8.x86_64) 8.5'",
+    },
+    "oracle-7": {
+        "yum_install_cmd": "yum install http://mirror.centos.org/centos/7/os/x86_64/Packages/kernel-3.10.0-1160.el7.x86_64.rpm -y",
+        "grub_cmd": "grub2-set-default 'Oracle Linux Server 7.9, with Linux 3.10.0-1160.el7.x86_64'",
+    },
+    "oracle-8.4": {
+        "yum_install_cmd": "yum install https://vault.centos.org/centos/8.4.2105/BaseOS/x86_64/os/Packages/kernel-core-4.18.0-305.25.1.el8_4.x86_64.rpm -y",
+        "grub_cmd": "",
+    },
+    # Install CentOS 8.5 kernel
+    "oracle-8.6": {
+        "yum_install_cmd": "yum install https://vault.centos.org/centos/8.5.2111/BaseOS/x86_64/os/Packages/kernel-core-4.18.0-348.7.1.el8_5.x86_64.rpm -y",
+        "grub_cmd": "grub2-set-default 'CentOS Linux (4.18.0-348.7.1.el8_5.x86_64) 8'",
+    },
+}
+
+
+def install_custom_kernel(shell):
+    """
+    Install CentOS kernel on Oracle Linux and vice versa to mimic the custom
+    kernel that is not signed by the running OS official vendor.
+    """
+    get_system_release = platform.platform()
+
+    yum_install_cmd, grub_cmd = DISTRO_KERNEL_MAPPING[get_system_release].values()
```

Comment:
```
14:56:03                 out:     def install_custom_kernel(shell):
14:56:03                 out:         """
14:56:03                 out:         Install CentOS kernel on Oracle Linux and vice versa to mimic the custom
14:56:03                 out:         kernel that is not signed by the running OS official vendor.
14:56:03                 out:         """
14:56:03                 out:         get_system_release = platform.platform()
14:56:03                 out:     
14:56:03                 out: >       yum_install_cmd, grub_cmd = DISTRO_KERNEL_MAPPING[get_system_release].values()
14:56:03                 out: E       KeyError: 'Linux-4.18.0-305.25.1.el8_4.x86_64-x86_64-with-centos-8.4.2105'
14:56:03                 out: 
14:56:03                 out: test_cutsom_kernel.py:43: KeyError
```

The problem is that `platform.platform()` returns more than just "centos-8". It's somehow simmilar to `uname -a` command.

Reply:
@r0x0d maybe We can try to use/copy this: https://github.com/oamg/convert2rhel/pull/619/files#diff-c87c5173e88bb0f60b00107f4586304f4869a976a4f998425efe3481c0db9f56R243-R287

Just copy paste it into this file and once @danmyway PR is merged we can remove it and use it just as a pytest fixture?

WDYT? @danmyway @r0x0d ?

Reply:
Oh :facepalm:. You're correct, @kokesak. 

I don't mind copying this over and later opening a PR to turn into a fixture. But since his code is in a `conftest.py`, I think I can apply the same here and use it as a fixture right now... Probably we can get away with merge conflicts if I place the fixture in the right place. 

Let me try.

Reply:
Yeah, why not. Lte's try it

Reply:
Also, @kokesak, not sure on what to do with the clean_up_custom_kernel function... Any ideas?

Reply:
@r0x0d, before the installation of different kernel I would try to save to some variable the currently loaded kernel, because we will want to get back to it. Maybe try to call `grubby --default-kernel` and save it. Then procceed with the test installing the new kernels.
When the test finishes, set the default krenel back to the original that it is saved inside the variable and do the reboot using the `tmt-reboot` again. Lastly remove the newly installed kernel (something like `yum undo last` or like that have the systsem in same state as before the test)

Reply:
Cool, I wll try that. Thanks!